receiver/receiver.go

```diff
@@ -20,6 +20,29 @@ import (
 	_ "github.com/open-telemetry/opentelemetry-service/internal/compression/grpc" // load in supported grpc compression encodings
 )
 
+// Host represents the entity where the receiver is being hosted. It is used to
+// allow communication between the receiver and its host.
+type Host interface {
+	// Context returns a context provided by the host to be used on the receiver
+	// operations.
+	Context() context.Context
+
+	// AsyncErrorChannel returns a channel used by a receiver to notify its
+	// host about any fatal error (that requires re-starting the receiver) that
+	// that happen after its start function has returned.
+	AsyncErrorChannel() chan<- error
+
+	// OkToIngest returns true when the receiver can inject the received data
+	// into the pipeline and false when it should drop the data and report
+	// error to the client.
+	//
+	// Receivers are expected to return the proper error (according to their
+	// protocol) when this function return false. For example an HTTP based
+	// receiver should return 503 (Service Unavailable) if OkToIngest returns
+	// false.
+	OkToIngest() bool
```

Comment:
I cannot see OkToIngest() used anywhere except tests. Is this coming in a future PR?

Reply:
Yes, per description of #76 it will be the second item.